packages/maskbook/src/web3/suspends/getERC721TokenDetailed.ts

```diff
@@ -31,14 +22,18 @@ async function ERC721TokenDetailed_(
         symbol: symbol ?? token?.symbol ?? '',
         baseURI: baseURI ?? token?.baseURI ?? '',
     } as ERC721TokenDetailed
-    cache.set(address, token_)
-    return
+
+    if (!cache.has(address)) cache.set(address, token_)
 }
 
-export function useAsyncERC721TokenDetailed(address: string, token?: Partial<ERC721TokenDetailed>) {
```

Comment:
I don't think to integrate with suspense must rename useXXX to getXXX which cause react hooks should be moved out which breaks our code structure, check https://git.io/JtWVq

Reply:
Generally, it doesn't matter what it is named. But if you use hooks in it, it must be a hook too.